src/libsyntax/parse/parser.rs

```diff
@@ -1052,7 +1052,7 @@ impl<'a> Parser<'a> {
 
         // examine next token to decide to do
         if self.eat_keyword_noexpect(keywords::Proc) {
-            self.parse_proc_type(lifetime_defs)
+            self.fatal("`proc` keyword isn't used for the moment");
```

Comment:
It's ok to remove this specific handling of the `Proc` keyword, an error should show up regardless somewhere in the parser eventually.